js/app/downloader.js

```diff
@@ -270,7 +270,8 @@ const SubscriptionDownloader = new Lang.Class({
             let manifest_uri = Soup.URI.new(uri);
 
             let request = new FileDownloadRequest(manifest_uri, new_manifest_file);
-            return this._file_downloader.download_file_promise(request, cancellable).catch(function (error) {
+            return this._file_downloader.download_file_promise(request, cancellable).catch((error) => {
+                this._app.release();
```

Comment:
It seems like you should make a separate `.catch` clause and tack it on to the very end. If I understand the code correctly, then it seems like this will not release the app if an error occurs in the following `.then` clause.

Maybe something like this for the final clause:
```js
.then(updated => {
    this._app.release();
    this._ongoing_downloads.delete(subscription_id);
    return updated;
}, error => {
    logError(error);
    this._app.release();
});
```